### ISSUE TITLE
Scaling options

### ISSUE THREAD

**Issue by trudnorx** [moonlight-stream/moonlight-qt#1002]
Multiple scaling options should exist:

* Stretch
* Maintain aspect ratio
* Zoom (customizable zoom level with hotkeys)

**Comment by rcz01** [comment]
At least when windowed, window size should resize so content is scaled 100%. Currently it starts with random size and everything is blurred. That is really annoying when using, e.g. 1080p on 4k screen. I use Moonlight with Sunshine in VMs as much faster alternative to RDP and this blur is driving me mad.
It should be a really simple change - window is already resized to some previous size, so code it there. Only change is not to change to previous one, but to size of the stream.

**Comment by tsgrissom** [comment]
Agreed! I'd like to reduce these horizontal black bars on the Steam Deck which seem like they could be remediated by some aspect ratio options within Moonlight.

**Comment by t9999clint** [comment]
I'd also like to mention the option for different scaling algorithms.
Biliniar, Nearest Neighbour, Lancos, HQ, FSR, Anime4K, etc...

Currently I use gamescope to do this, but that's kind of a pain in the arse to use. So I'd much rather prefer a built in solution.

**Comment by PHLAK** [comment]
Having FSR (or similar) would be nice. My PC is 1440p and I often stream to my laptop connected to my 4K TV. While I can configure some games to render at 4K it's not possible without jumping through some hoops (i.e. enabling Virtual Super Resolution in AMD Adrenaline _and_ setting my desktop resolution to 3840 × 2160 _and_ setting the game resolution to match). However, if I could stream 1440 and upscale it on the client device this would be a perfect compromise.

**Comment by beren12** [comment]
I'd be happy if I could stretch my 16:10 remote screen to the ipad's 4:3. I don't always care if the picture is stretched. Gamestream with a custom rez on the ipad would do that but sunshine doesn't

**Comment by chopin1998** [comment]
any update?
